client/types.go

```diff
@@ -0,0 +1,109 @@
+package client
+
+import (
+	"net/url"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	insightsInsertURL = "https://insights-collector.newrelic.com/v1/accounts"
+	insightsQueryURL  = "https://insights-api.newrelic.com/v1/accounts"
+
+	// DefaultBatchTimeout is the amount of time to submit batches even if the event count hasn't been hit
+	DefaultBatchTimeout = 1 * time.Minute
+	// DefaultBatchEventCount is the maximum number of events before sending a batch (fuzzy)
+	DefaultBatchEventCount = 950
+	// DefaultWorkerCount is the number of background workers consuming and sending events
+	DefaultWorkerCount = 1
+
+	// DefaultInsertRequestTimeout is the amount of seconds to wait for a insert response
+	DefaultInsertRequestTimeout = 10 * time.Second
+	// DefaultQueryRequestTimeout is the amount of seconds to wait for a query response
+	DefaultQueryRequestTimeout = 20 * time.Second
+
+	// DefaultRetries is how many times to attempt the query
+	DefaultRetries = 3
+	// DefaultRetryWaitTime is the amount of seconds between query attempts
+	DefaultRetryWaitTime = 5 * time.Second
+)
+
+// Compression to use during transport.
+type Compression int32
+
+// Supported / recognized types of compression
+const (
+	None    Compression = iota
+	Deflate Compression = iota
+	Gzip    Compression = iota
+	Zlib    Compression = iota
+)
+
+// Client is the building block of the insert and query clients
+type Client struct {
+	URL            *url.URL
+	Logger         *log.Logger
+	RequestTimeout time.Duration
+	RetryCount     int
+	RetryWait      time.Duration
+}
+
+// InsertClient contains all of the configuration required for inserts
+type InsertClient struct {
+	InsertKey   string
+	eventQueue  chan []byte
+	eventTimer  *time.Timer
+	flushQueue  chan bool
+	WorkerCount int
+	BatchSize   int
+	BatchTime   time.Duration
+	Compression Compression
+	Client
+	Statistics
+}
+
+// Statistics about the inserted data
+type Statistics struct {
```

Comment:
non thread safe, but you didn't work on this ;)

Reply:
Not going to mess with `Statistics` on this PR, but you're correct...